apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js

```diff
@@ -0,0 +1,555 @@
+  type: REMOVE_LEVEL,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition
+});
+
+export const chooseLevel = (
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  variant,
+  value
+) => ({
+  type: CHOOSE_LEVEL,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  variant,
+  value
+});
+
+export const addVariant = (
+  activityPosition,
+  activitySectionPosition,
+  levelPosition
+) => ({
+  type: ADD_VARIANT,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition
+});
+
+export const removeVariant = (
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  levelId
+) => ({
+  type: REMOVE_VARIANT,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  levelId
+});
+
+export const setActiveVariant = (
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  id
+) => ({
+  type: SET_ACTIVE_VARIANT,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  id
+});
+
+export const setField = (
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  modifier
+) => ({
+  type: SET_FIELD,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  modifier
+});
+
+export const reorderLevel = (
+  activityPosition,
+  activitySectionPosition,
+  originalLevelPosition,
+  newLevelPosition
+) => ({
+  type: REORDER_LEVEL,
+  activityPosition,
+  activitySectionPosition,
+  originalLevelPosition,
+  newLevelPosition
+});
+
+export const moveLevelToActivitySection = (
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  newActivitySectionPosition
+) => ({
+  type: MOVE_LEVEL_TO_ACTIVITY_SECTION,
+  activityPosition,
+  activitySectionPosition,
+  levelPosition,
+  newActivitySectionPosition
+});
+
+export const addLevel = (activityPosition, activitySectionPosition, level) => ({
+  type: ADD_LEVEL,
+  activityPosition,
+  activitySectionPosition,
+  level
+});
+
+export const moveActivity = (activityPosition, direction) => ({
+  type: MOVE_ACTIVITY,
+  activityPosition,
+  direction
+});
+
+export const moveActivitySection = (
+  activityPosition,
+  activitySectionPosition,
+  direction
+) => ({
+  type: MOVE_ACTIVITY_SECTION,
+  activityPosition,
+  activitySectionPosition,
+  direction
+});
+
+export const removeActivity = activityPosition => ({
+  type: REMOVE_ACTIVITY,
+  activityPosition
+});
+
+export const removeActivitySection = (
+  activityPosition,
+  activitySectionPosition
+) => ({
+  type: REMOVE_ACTIVITY_SECTION,
+  activityPosition,
+  activitySectionPosition
+});
+
+export const setActivity = (
+  activitySectionPosition,
+  oldActivityPosition,
+  newActivityPosition
+) => ({
+  type: SET_ACTIVITY,
+  activitySectionPosition,
+  oldActivityPosition,
+  newActivityPosition
+});
+
+export const addTip = (activityPosition, activitySectionPosition, tip) => ({
+  type: ADD_TIP,
+  activityPosition,
+  activitySectionPosition,
+  tip
+});
+
+function updateActivityPositions(activities) {
```

Comment:
Do we need to maintain these `position` values in the redux state if these are already stored in Arrays? It seems like we could simplify some of the logic by not doing so, since the logic for moving things around already updates the order within the array.

If this is needed on the backend, generating position values could be done by the rails controller.

Reply:
I think this could be a good thing to look at as well but since the Activities Editor is working with this I think that should probably be follow up work.